apps/prairielearn/src/lib/editors.ts

```diff
@@ -180,8 +180,16 @@ export function getUniqueNames({
   }
 }
 
-interface BaseEditorOptions {
-  locals: Record<string, any>;
+interface BaseEditorOptions<ResLocals = object> {
+  locals: {
+    authz_data: Record<string, any>;
+    course: Course;
+    user: User;
+  } & ResLocals;
+}
+
+interface BaseEditorOptionsInternal extends BaseEditorOptions {
+  description: string;
```

Comment:
In practice this doesn't add any type safety to calls since `res.locals` is always a big unknown thing, but it does give us slightly more guarantees when editors are used.